sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/basicLogicalOperators.scala

```diff
@@ -270,6 +279,8 @@ case class Union(
     }
   }
 
+  override def metadataOutput: Seq[Attribute] = children.flatMap(_.metadataOutput)
```

Comment:
not sure about Union. It only propagates the output for the first child, and `ResolveMissingReferences` doesn't support it either.

Reply:
Same to `Intersect` `Except`